File: farmersmk-marketplace-service/db-connection-example.java
// Example PostgreSQL connection in Java using JDBC
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.SQLException;

public class DbConnectionExample {
    public static void main(String[] args) {
        String url = "jdbc:postgresql://db:5432/farmersmkdb";
        String user = "farmersmk";
        String password = "farmersmkpass";

        try (Connection conn = DriverManager.getConnection(url, user, password)) {
            if (conn != null) {
                System.out.println("Connected to the database!");
            } else {
                System.out.println("Failed to make connection!");
            }
        } catch (SQLException e) {
            System.out.println(e.getMessage());
        }
    }
}

File: farmersmk-frontend-web/db-connection-example.js
// Example PostgreSQL connection in Node.js using 'pg' package
const { Client } = require('pg');

const client = new Client({
  host: 'db',
  port: 5432,
  user: 'farmersmk',
  password: 'farmersmkpass',
  database: 'farmersmkdb',
});

client.connect()
  .then(() => console.log('Connected to the database!'))
  .catch(err => console.error('Connection error', err.stack))
  .finally(() => client.end());
